datalad_next/create_sibling_webdav.py

```diff
@@ -422,41 +429,107 @@ def _create_sibling_webdav(
     # simplify downstream logic, export yes or no
     export_storage = 'export' in storage_sibling
 
+    existing_siblings = [r[1] for r in
+                         _yield_ds_w_matching_siblings(ds,
+                                                       (name, storage_name),
+                                                       recursive=False)
+                         ]
+
     if storage_sibling != 'no':
-        yield from _create_storage_sibling(
-            ds,
-            url,
-            storage_name,
-            existing,
-            credential,
-            export=export_storage,
-        )
+        if storage_name in existing_siblings:
+            yield from maybe_skip_sibling(credential, credential_name, ds,
+                                          existing, export_storage, name,
+                                          storage_name, storage_sibling,
+                                          url, 'storage')
+        else:
```

Comment:
TBH: I do not like this pattern.

Every call is now duplicated, because one decision re `existing` (namely is there a conflict) is tested outside, and the rest inside `maybe_skip_sibling`.

`maybe_skip_sibling` is also maybe-skip-or-reconfigure`. And carries a huge signature only to be able to set `reconfigure=True`.

Why not 

- RF into `_skip_sibling(name, existing_siblings, existing) -> bool`
- replace `if (storage_)name in existing_siblings` with `if _skip_sibling((storage_)name, existing_sibling, existing)`
- perform all result yielding here
- perform all calls to `_create(_storage)_sibling` here, and once
- set their `reconfigure` to `storage_name in existing_siblings` (cached)

should half the code

Reply:
Done